### PR TITLE
pgpPubkeyFingerprint for Sequoia

### PR DESCRIPTION
**Pull request by nwalfield**
This builds on https://github.com/rpm-software-management/rpm/pull/2172 .

One concern that I have is that the internal parser only supports binary keys, but Sequoia supports ASCII-armor keys.  Should Sequoia explicitly not support ASCII armor keys?

### CODE REVIEW

**Comment by pmatilai** [comment]
Hmm. I'd say limit to binary only for consistency's sake, ie callers are required to explicitly "convert" (dearmor?) ASCII armors to binary before use in these other APIs. We can always reconsider when the internal parser stops mattering.

**Comment by nwalfield** [comment]
FWIW, I don't think rpm has a dearmor API, does it?

File: rpmio/rpmpgp_sequoia.c [hunk 1]
```diff
@@ -39,6 +39,9 @@ W(rpmRC, pgpVerifySignature,
 W(int, pgpPubkeyKeyID,
   (const uint8_t * pkt, size_t pktlen, pgpKeyID_t keyid),
   (pkt, pktlen, keyid))
+W(int, pgpPubkeyFingerprint,
+  (const uint8_t * pkt, size_t pktlen, uint8_t **fp, size_t *fplen),
+  (pkt, pktlen, fp, fplen))
```
Comment (pmatilai) [inline rpmio/rpmpgp_sequoia.c:44]: Doh, I would've added this already but totally forgot that we need such a change on rpm side too. And that was even by my request :facepalm::laughing:

Comment (nwalfield) [reply rpmio/rpmpgp_sequoia.c:44]: No worries!

**Comment by pmatilai** [comment]
> FWIW, I don't think rpm has a dearmor API, does it?

It does - pgpReadPkts() and pgpParsePkts(). It's kinda funny as rpm --import *only* accepts ASCII armored keys.

**Comment by nwalfield** [comment]
I've merged the rpm-sequoia support here: https://github.com/rpm-software-management/rpm-sequoia/pull/8

(As requested, it only accepts binary certificates.)

**Comment by nwalfield** [comment]
> > FWIW, I don't think rpm has a dearmor API, does it?
> 
> It does - pgpReadPkts() and pgpParsePkts(). It's kinda funny as rpm --import _only_ accepts ASCII armored keys.

Indeed.  Thanks.

**Comment by pmatilai** [comment]
Thanks for the quick response + testcase!

**Comment by pmatilai** [comment]
Our CI seems to be stuck, builds complete on Semaphore just fine but the completion message never comes through to GH. Using big red button override...

**Pull request merged**